src/common.rs

```diff
@@ -0,0 +1,492 @@
+    client: &mut Client,
+    keystore: FilesystemKeyStore<StdRng>,
+) -> Result<(miden_client::account::Account, SecretKey), ClientError> {
+    let mut init_seed = [0_u8; 32];
+    client.rng().fill_bytes(&mut init_seed);
+
+    let key_pair = SecretKey::with_rng(client.rng());
+    let builder = AccountBuilder::new(init_seed)
+        .account_type(AccountType::RegularAccountUpdatableCode)
+        .storage_mode(AccountStorageMode::Public)
+        .with_auth_component(RpoFalcon512::new(key_pair.public_key()))
+        .with_component(BasicWallet);
+    let (account, seed) = builder.build().unwrap();
+    client.add_account(&account, Some(seed), false).await?;
+    keystore
+        .add_key(&AuthSecretKey::RpoFalcon512(key_pair.clone()))
+        .unwrap();
+
+    Ok((account, key_pair))
+}
+
+/// Creates a public immutable network smart contract account from the provided MASM code.
+///
+/// This function compiles the provided account code into a contract with immutable code,
+/// public storage mode, and no authentication requirements. The contract is initialized
+/// with a zero-value storage slot.
+///
+/// # Arguments
+///
+/// * `client` - A mutable reference to the Miden client
+/// * `account_code` - The MASM code that defines the contract's behavior
+///
+/// # Returns
+///
+/// Returns a `Result` containing a tuple of the created contract `Account` and its seed `Word`,
+/// or a `ClientError` if contract creation fails.
+pub async fn create_network_account(
+    client: &mut Client,
+    account_code: &str,
+) -> Result<(Account, Word), ClientError> {
+    let assembler: Assembler = TransactionKernel::assembler().with_debug_mode(true);
+
+    let counter_component = AccountComponent::compile(
+        account_code.to_string(),
+        assembler.clone(),
+        vec![StorageSlot::Value([
+            Felt::new(0),
+            Felt::new(0),
+            Felt::new(0),
+            Felt::new(0),
+        ])],
+    )
+    .unwrap()
+    .with_supports_all_types();
+
+    let mut init_seed = [0_u8; 32];
+    client.rng().fill_bytes(&mut init_seed);
+
+    let (counter_contract, counter_seed) = AccountBuilder::new(init_seed)
+        .account_type(AccountType::RegularAccountImmutableCode)
+        .storage_mode(AccountStorageMode::Network)
+        .with_auth_component(auth::NoAuth)
+        .with_component(counter_component.clone())
+        .build()
+        .unwrap();
+
+    Ok((counter_contract, counter_seed))
+}
+
+pub async fn create_public_account(
+    client: &mut Client,
+    account_code: &str,
+) -> Result<(Account, Word), ClientError> {
+    let assembler: Assembler = TransactionKernel::assembler().with_debug_mode(true);
+
+    let counter_component = AccountComponent::compile(
+        account_code.to_string(),
+        assembler.clone(),
+        vec![StorageSlot::Value([
+            Felt::new(0),
+            Felt::new(0),
+            Felt::new(0),
+            Felt::new(0),
+        ])],
```

Comment:
see my comment in the original PR: https://github.com/0xMiden/project-template/pull/4#discussion_r2250079452